src/react/Routes.jsx

```diff
@@ -1,67 +1,63 @@
+
+            <Route exact path="/create-location" component={LocationEditor} />
+            <Route path="/locations/:locationName/edit" component={LocationEditor} />
+
+            <Route path='/accounts/:accountName?' component={Accounts} />
+            <Route path="/create-account" component={AccountCreate} />
+
+            <Route path={['/buckets/:bucketName?', '/buckets/:bucketName/objects', '/create-bucket']} component={DataBrowser} />
 
-    const dispatch: DispatchAPI<Action> = useDispatch();
+            <Route path={['/create-workflow', '/workflows/:ruleId?']} component={Workflows} />
+
+            <Route path="*" component={NoMatch}/>
+        </Switch>
+    );
+}
+
+function Routes() {
     return (
         <RouteContainer>
             <NavbarContainer>
-                <Navbar
-                    rightActions={[
-                        {
-                            icon: <i className="fas fa-user"/>,
-                            text: `${userName}`,
-                            type: 'dropdown',
-                            items: [{ label: 'Log out', onClick: () => dispatch(signout()) } ],
-                        }]
-                    }
-                    tabs={[
-                        {
-                            link: <Link to="/accounts">Accounts</Link>,
-                            selected: !!matchPath(pathname, { path: '/accounts/:accountName?' }) ||
-                                      !!matchPath(pathname, { path: '/create-account' }),
-                        },
-                        {
-                            link: <Link to="/buckets">Data Browser</Link>,
-                            selected: !!matchPath(pathname, { path: '/buckets' }) ||
-                                      !!matchPath(pathname, { path: '/create-bucket' }),
-                        },
-                        {
-                            link: <Link to="/workflows">Workflow</Link>,
-                            selected: !!matchPath(pathname, { path: '/workflows' }),
-                        },
-                    ]}
-                />
+                <Navbar/>
             </NavbarContainer>
             <Switch>
-                <Route exact path="/" render={() => <Redirect to="/accounts" />}/>
-
-                <Route exact path="/create-location" component={LocationEditor} />
-                <Route path="/locations/:locationName/edit" component={LocationEditor} />
-
-                <Route path='/accounts/:accountName?' component={Accounts} />
-                <Route path="/create-account" component={AccountCreate} />
-
-                <Route path={['/buckets/:bucketName?', '/buckets/:bucketName/objects', '/create-bucket']} component={DataBrowser} />
-                <Route path={['/create-workflow', '/workflows/:ruleId?']} component={Workflows} />
-                <Route path="*" component={NoMatch}/>
+                <Route exact path="/login/callback" component={LoginCallback}/>
```

Comment:
I have not found a clean way to deal with:
- the authorization server redirecting the user back ui.zenko.local with an authorization code
Indeed, if I do not specify `redirectUrl`:
1) auth server will redirect to `http://ui.zenko.local/accounts/test-account-1?state=<STATE>&session_state=<SESSION_STATE>&code=<CODE>`. 
2) solutions-navbar--authenticated will fire
3) UI will load the clients and requests will fail because shell UI redirect to http://ui.zenko.local/accounts/test-account-1 (onSignIn is fired) https://github.com/scality/metalk8s/blob/development/2.9/shell-ui/src/index.js#L158

There is no UI visible errors (since the redirect step 3 clean errors up with the redirect) but it seems wrong to me to redirect to a private route. What do you think @JBWatenbergScality ?

Reply:
My approach would be the following : 
- Use `http://ui.zenko.local/` as the redirectUrl
- When trying to access a private route while being unauthenticated (such as `/accounts/test-account-1`), the navbar will save this url and redirect the user to the auth server 
- Once authenticated, the auth server will redirect to `http://ui.zenko.local/?state=<STATE>&session_state=<SESSION_STATE>&code=<CODE>`
- The navbar will then redirect back to `http://ui.zenko.local/accounts/test-account-1` (https://github.com/scality/metalk8s/blob/development/2.9/shell-ui/src/index.js#L158 indeed)
- `solutions-navbar--authenticated` is triggered giving access to the protected route

Reply:
The only issue I was facing is `Once authenticated, the auth server will redirect to http://ui.zenko.local/?state=<STATE>&session_state=<SESSION_STATE>&code=<CODE>` step, where my user was authenticated (userManager.events.addUserLoaded)  and clients (S3, API..) were initializing before being redirected by https://github.com/scality/metalk8s/blob/development/2.9/shell-ui/src/index.js#L158 which leads to some canceled calls.